### PR TITLE
feat: booking service date

### PR DESCRIPTION
**Pull request by mcksyrek**

### CODE REVIEW

File: src/main/java/com/bookingplatform/server/timetable/TimetableService.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    private List<Integer> _getReservationsHours(Timetable timetable) {
+        List<Integer> reservedHours = timetable.getReservations()
+                .stream()
+                .map(reservation -> reservation.getHour())
+                .collect(Collectors.toCollection(ArrayList::new));
+        return reservedHours;
+    }
+
+    private Timetable _createNewTimetable(Long id, String date) {
+        Timetable newTimetable =  new Timetable();
```
Comment (ArtemLunt) [inline src/main/java/com/bookingplatform/server/timetable/TimetableService.java:62]: Extra whitespace

File: src/main/java/com/bookingplatform/server/timetable/TimetableService.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        newTimetable.setOffer(
+                offerRepository.findById(id)
+                        .orElseThrow(() ->
+                                new ResponseStatusException(
+                                        HttpStatus.NOT_FOUND,
+                                        ExceptionConstants.OFFER_NOT_FOUND)
+                        )
+        );
+        newTimetable.setDate(date);
+        newTimetable.setReservations(new ArrayList<>());
```
Comment (ArtemLunt) [inline src/main/java/com/bookingplatform/server/timetable/TimetableService.java:72]: Pls consider the usage of the [wither](https://projectlombok.org/features/With).

File: src/main/java/com/bookingplatform/server/timetable/TimetableService.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    private Timetable _getTimetableByOfferIdAndDate(Long id, String date) {
+        Optional<Timetable> selectedTimetable = timetableRepository.findByOfferIdAndDate(id, date);
+        return selectedTimetable.orElse(_createNewTimetable(id, date));
+    }
+
+    private List<Integer> _getReservationsHours(Timetable timetable) {
+        List<Integer> reservedHours = timetable.getReservations()
+                .stream()
+                .map(reservation -> reservation.getHour())
```
Comment (ArtemLunt) [inline src/main/java/com/bookingplatform/server/timetable/TimetableService.java:56]: We can make it shorter with the method reference: `Reservation::getHours`.

File: src/main/java/com/bookingplatform/server/timetable/TimetableService.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    public List<Integer> getReservationHoursFromSelectedTimetable(Long id, String date) {
+        return _getReservationsHours(_getTimetableByOfferIdAndDate(id, date));
+    }
+
+    public Timetable addReservation(Long id, String date, Integer duration,  Reservation reservation) {
+        Timetable selectedTimetable = _getTimetableByOfferIdAndDate(id, date);
+
+        List<Reservation> reservationDurationList = new ArrayList<Reservation>();
+        while (duration > 0) {
+            duration = duration - 1;
```
Comment (ArtemLunt) [inline src/main/java/com/bookingplatform/server/timetable/TimetableService.java:33]: Why not for?

**Review by ArtemLunt — CHANGES_REQUESTED** [review]

File: src/main/java/com/bookingplatform/server/timetable/TimetableService.java (hunk 2 above)
Comment (mcksyrek) [reply src/main/java/com/bookingplatform/server/timetable/TimetableService.java:72]: wow that is what i have been looking for!

File: src/main/java/com/bookingplatform/server/reservation/Reservation.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+    @With
+    private Integer hour;
+    private String product;
+
+    @ManyToOne
+    @JoinColumn(name = "timetable_id")
+    @JsonBackReference
+    @With
```
Comment (ArtemLunt) [inline src/main/java/com/bookingplatform/server/reservation/Reservation.java:24]: You can simply mark the class with the `@Wither` annotation

File: src/main/java/com/bookingplatform/server/timetable/TimetableService.java (hunk 2 above)
Comment (ArtemLunt) [reply src/main/java/com/bookingplatform/server/timetable/TimetableService.java:74]: @mcksyrek so why r u still using setters here? :^)

**Review by ArtemLunt — CHANGES_REQUESTED** [review]

**Review by ArtemLunt — APPROVED** [review]

**Pull request merged**